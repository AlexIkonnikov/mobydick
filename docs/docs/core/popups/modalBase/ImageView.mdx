Reusable ImageView component for ModalBase

## `Props`

### <div class="label required basic">Required</div>**`image`**

| TYPE         |
|:-------------|
| ReactElement |

Image for ImageView

### `imageStyles`

| TYPE                                                             |
|:-----------------------------------------------------------------|
| [ViewStyle](https://reactnative.dev/docs/view-style-props#props) |

ImageStyles for ImageView
