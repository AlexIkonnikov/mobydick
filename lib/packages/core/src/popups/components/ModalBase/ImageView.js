"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const useStyles_1 = __importDefault(require("../../../styles/hooks/useStyles"));
const View_1 = __importDefault(require("../../../basic/components/View/View"));
const stylesCreate_1 = __importDefault(require("./stylesCreate"));
const ImageView = props => {
    const [styles] = (0, useStyles_1.default)(stylesCreate_1.default);
    return (<View_1.default style={[styles.imageView, props.imageStyles]}>{props.image}</View_1.default>);
};
exports.default = ImageView;
