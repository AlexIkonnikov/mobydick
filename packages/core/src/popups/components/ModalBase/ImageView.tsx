import React, {FC, ReactElement} from 'react';
import {ViewStyle} from 'react-native';

import useStyles from '../../../styles/theme/hooks/useStyles';
import View from '../../../basic/components/View/View';

import stylesCreate from './stylesCreate';

interface IProps {
  image: ReactElement;
  imageStyles?: ViewStyle | ViewStyle[];
}

const ImageView: FC<IProps> = props => {
  const [styles] = useStyles(stylesCreate);

  return (
    <View style={[styles.imageView, props.imageStyles]}>{props.image}</View>
  );
};

export default ImageView;
