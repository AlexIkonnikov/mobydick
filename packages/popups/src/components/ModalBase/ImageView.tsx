import React, {FC, ReactElement} from 'react';
import {View} from '@npm/mobydick-core';
import {useStyles} from '@npm/mobydick-styles';
import {ViewStyle} from 'react-native';

import stylesCreate from './stylesCreate';

interface IProps {
  image: ReactElement;
  imageStyles?: ViewStyle | ViewStyle[];
}

const ImageView: FC<IProps> = props => {
  const [styles] = useStyles(stylesCreate);

  return (
    <View style={[styles.imageView, props.imageStyles]}>{props.image}</View>
  );
};

export default ImageView;
