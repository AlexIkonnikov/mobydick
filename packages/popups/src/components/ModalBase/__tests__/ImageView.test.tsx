import {render} from '@testing-library/react-native';
import React from 'react';
import {View} from '@npm/mobydick-core';

import ImageView from '../ImageView';

describe('@npm/mobydick-popups/ImageView', () => {
  it('should renders correctly', () => {
    const {toJSON} = render(<ImageView image={<View />} />);

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly with styles', () => {
    const {toJSON} = render(
      <ImageView image={<View />} imageStyles={{flex: 1}} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
