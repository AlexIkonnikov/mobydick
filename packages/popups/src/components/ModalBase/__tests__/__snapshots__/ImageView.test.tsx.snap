// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/ImageView should renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "marginTop": 36,
      },
      undefined,
    ]
  }
>
  <View />
</View>
`;

exports[`@npm/mobydick-popups/ImageView should renders correctly with styles 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "marginTop": 36,
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View />
</View>
`;
